astroid/nodes/node_classes.py

```diff
@@ -4489,12 +4489,12 @@ class MatchSingleton(Pattern):
     def __init__(
         self,
         *,
-        value: Literal[True, False, None],
+        value: Literal[True, False, None],  # type: ignore[assignment] # See https://github.com/python/mypy/pull/10389
```

Comment:
Why is the `type: ignore` comment necessary here at all? I didn't notice any errors without it.

Reply:
That was indeed misplaced. Should have been in `rebuilder`. See latest commit, `node.value` is `Optional[bool]`.